runtimes/rust/tests/inspected/mod.rs

```diff
@@ -0,0 +1,378 @@
+    blockchain::{
+        config::{GenesisConfigBuilder, InstanceInitParams},
+        Blockchain, BlockchainMut, Schema as CoreSchema,
+    },
+    crypto::Hash,
+    helpers::{generate_testnet_config, Height, ValidatorId},
+    merkledb::{ObjectHash, Patch, Snapshot},
+    messages::{AnyTx, Verified},
+    runtime::{
+        migrations::{InitMigrationError, MigrationScript},
+        versioning::Version,
+        CallInfo, ExecutionContext, InstanceSpec, InstanceStatus, Mailbox, Runtime,
+        WellKnownRuntime,
+    },
+};
+use exonum_derive::{exonum_interface, BinaryValue, ServiceDispatcher, ServiceFactory};
+use futures::Future;
+use serde_derive::*;
+
+use std::{
+    collections::BTreeMap,
+    sync::{Arc, Mutex},
+};
+
+use exonum_rust_runtime::{
+    ArtifactId, CallContext, DefaultInstance, ExecutionError, InstanceId, RustRuntime, Service,
+    ServiceFactory, SnapshotExt, SUPERVISOR_INSTANCE_ID,
+};
+
+pub fn execute_transaction(
+    blockchain: &mut BlockchainMut,
+    tx: Verified<AnyTx>,
+) -> Result<(), ExecutionError> {
+    let tx_hash = tx.object_hash();
+
+    let (block_hash, patch) = create_block_with_transactions(blockchain, vec![tx]);
+    blockchain
+        .commit(patch, block_hash, vec![], &mut BTreeMap::new())
+        .unwrap();
+
+    let snapshot = blockchain.snapshot();
+    let schema = CoreSchema::new(&snapshot);
+    let location = schema.transactions_locations().get(&tx_hash).unwrap();
+    schema.transaction_result(location).unwrap()
+}
+
+pub fn create_block_with_transactions(
+    blockchain: &mut BlockchainMut,
+    transactions: Vec<Verified<AnyTx>>,
+) -> (Hash, Patch) {
+    let tx_hashes = add_transactions_into_pool(blockchain, transactions);
+
+    let height = {
+        let snapshot = blockchain.snapshot();
+        CoreSchema::new(&snapshot).next_height()
+    };
+
+    blockchain.create_patch(
+        ValidatorId::zero().into(),
+        height,
+        &tx_hashes,
+        &mut BTreeMap::new(),
+    )
+}
+
+pub fn create_genesis_config_builder() -> GenesisConfigBuilder {
+    let consensus_config = generate_testnet_config(1, 0)[0].clone().consensus;
+    GenesisConfigBuilder::with_consensus_config(consensus_config)
+}
+
+fn add_transactions_into_pool(
+    blockchain: &mut BlockchainMut,
+    txs: Vec<Verified<AnyTx>>,
+) -> Vec<Hash> {
+    blockchain
+        .merge({
+            let fork = blockchain.fork();
+            let mut schema = CoreSchema::new(&fork);
+            for tx in txs.clone() {
+                schema.add_transaction_into_pool(tx);
+            }
+            fork.into_patch()
+        })
+        .unwrap();
+
+    txs.into_iter().map(|x| x.object_hash()).collect()
+}
+
+#[derive(Debug, PartialEq)]
+pub enum RuntimeEvent {
+    Initialize,
+    Resume,
+    BeforeTransactions(Height, InstanceId),
+    DeployArtifact(ArtifactId, Vec<u8>),
+    StartAdding(InstanceSpec, Vec<u8>),
+    Migrate(ArtifactId, Version),
+    StartResuming(InstanceSpec, Vec<u8>),
```

Comment:
This variant looks a bit confusing being near the `Resume` variant. I'd suggest to rename them to `StartInstanceResuming` and `ResumeRuntime` correspondingly.